### PR TITLE
boards: xtensa: add heltec_wireless_stick_lite_v3

### PR DESCRIPTION
**Pull request by adolfogc**

### CODE REVIEW

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,208 @@
+.. heltec_wireless_stick_lite_v3:
+
+HelTec Wireless Stick Lite (V3)
+###############################
+
+Overview
+********
+
+HelTec Wireless Stick Lite (V3) is a development board with support for Wi-Fi, Bluetooth and LoRa designed, produced by HelTec Automation(TM). [1]_
```
Comment (sylvioalves) [inline boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:9]: I suggest this:  
```
HelTec Wireless Stick Lite (V3) is a development board with Wi-Fi, Bluetooth and LoRa support. It is designed and produced by HelTec Automation(TM).
```

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+   :alt: Board Name
+
+   Board Name (Credit: Chengdu HelTec Automation Technology Co., Ltd.)
+
+Hardware
+********
+
+The main hardware features are:
+
+- ESP32-S3FN8 (Xtensa® 32-bit LX7 dual core processor, five stage pipeline rack Structure, main frequency up to 240 MHz) as MCU
```
Comment (sylvioalves) [inline boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:23]: `as MCU` -> `and MCU`

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+- Semtech SX1262 LoRa node chip
+- Type-C USB interface with a complete voltage regulator, ESD protection, short circuit protection, RF shielding, and other protection measures (note: you need an USB-A to USB-C cable if you want power-up the board from USB).
+- Onboard SH1.25-2 battery interface, integrated lithium battery management system (charge and discharge management, overcharge protection, battery power detection, USB / battery power automatic switching).
+- Integrated WiFi, LoRa, Bluetooth three network connections, onboard Wi-Fi, Bluetooth dedicated 2.4GHz metal spring antenna, reserved IPEX (U.FL) interface for LoRa use.
```
Comment (sylvioalves) [inline boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:27]: Line below could be improved to:
```
- Integrated WiFi and Bluetooth interfaces with 2.4GHz metal spring antenna and reserved IPEX (U.FL) interface for LoRa use.
```

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+   to the board default configuration file.
+   ```
+   CONFIG_BOOTLOADER_MCUBOOT=y
+   ```
+
+Sysbuild
+========
+
+The sysbuild makes possible to build and flash all necessary images needed to
+bootstrap the board with the EPS32 SoC.
```
Comment (sylvioalves) [inline boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:91]: `EPS32S3 SoC.`

**Comment by adolfogc** [comment]
@sylvioalves Thank you for the feedback! Please see my last commit with the suggested improvements.

**Comment by marekmatej** [comment]
Hi @adolfogc thanks for your contribution! Please squash the commits to a single one.

File: boards/xtensa/heltec_wireless_stick_lite_v3/board_init.c [hunk 5]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2021 Instituto de Pesquisas Eldorado (eldorado.org.br)
+ * Copyright (c) 2023 The Zephyr Project Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gpio.h>
```
Comment (sylvioalves) [inline boards/xtensa/heltec_wireless_stick_lite_v3/board_init.c:8]: Please, add: 

```c
#include <zephyr/device.h>
#include <zephyr/init.h>
```

**Comment by sylvioalves** [comment]
@adolfogc, have you tested this board with `samples/drivers/lora/` sample? Does it fully work?

File: boards/xtensa/heltec_wireless_stick_lite_v3/board_init.c (hunk 5 above)
Comment (adolfogc) [reply boards/xtensa/heltec_wireless_stick_lite_v3/board_init.c:9]: Thanks! I've added the fix.

**Comment by adolfogc** [comment]
@marekmatej, @sylvioalves Thank you for your feedback! I have completed the changes as per the latest comments. I haven't tested the LoRa functionality, but I have now ran the samples. The result was the following: 1) no errors reported, 2) both the receive and the send calls block without further activity. For the test I loaded the sample firmware into two separate boards and inspected the serial output.

**Review by sylvioalves — DISMISSED** [review]

**Comment by sylvioalves** [comment]
> @marekmatej, @sylvioalves Thank you for your feedback! I have completed the changes as per the latest comments. I haven't tested the LoRa functionality, but I have now ran the samples. The result was the following: 1) no errors reported, 2) both the receive and the send calls block without further activity. For the test I loaded the sample firmware into two separate boards and inspected the serial output.

Thanks, that you be good to confirm whether LoRa is indeed working between boards. Let us know if/when you confirm that. Nice work!

**Comment by adolfogc** [comment]
@sylvioalves I made some adjustments to the LoRa device node and now the send driver sample runs successfully:

```
[00:00:00.605,000] <inf> lora_send: Data sent!
[00:00:01.904,000] <inf> lora_send: Data sent!
[00:00:03.204,000] <inf> lora_send: Data sent!
[00:00:04.503,000] <inf> lora_send: Data sent!
[00:00:05.803,000] <inf> lora_send: Data sent!
[00:00:07.102,000] <inf> lora_send: Data sent!
[00:00:08.402,000] <inf> lora_send: Data sent!
```

I also made some other improvements.

**Comment by sylvioalves** [comment]
@adolfogc, I keep think whether I2C and TWAI/CAN should really be added into default board's DTS as it has no peripherals/interface. Is there an I2C device or CAN driver at least?

**Comment by adolfogc** [comment]
@sylvioalves There is no I2C device present in the board, so I will remove it from the DTS. Regarding the CAN interface, I thought that the ESP32-S3 family had driver support already, is this not the case? Also, since there is no transceiver present in the board, should it be excluded from the DTS as well?

**Comment by adolfogc** [comment]
@sylvioalves I left the TWAI and I2C-0 nodes in the DTS, but disabled them, and added a comment indicating that they can be enabled in a board extension. I checked the DTSI for the ESP32-S3 SoC and there's a driver for TWAI. However, some comments in the driver's source code suggest that some stuff may be pending to implement(?). Is that what you meant w.r.t. the TWAI driver?

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+|    J2.10   |    GPIO37    |         ADC Control         |
+|    J2.11   |      TBD     |                             |
+|    J2.12   |    GPIO39    |                             |
+|    J2.13   |    GPIO40    |                             |
+|    J2.14   |    GPIO41    |                             |
+|    J2.15   |    GPIO42    |                             |
+|    J2.16   |    GPIO45    |                             |
+|    J2.17   |    GPIO46    |                             |
+|    J2.18   |   ADC1_CH0   | Battery Voltage Measurement |
+|    J2.19   |      GPIO3/I2C0_SCL     |       I2C-0 (optional*)  |
```
Comment (sylvioalves) [inline boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:72]: Hi, would you fix this alignment? I think it was correct in last commit.

File: boards/xtensa/heltec_wireless_stick_lite_v3/heltec_wireless_stick_lite_v3.dts [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	pinctrl-0 = <&i2c0_default>;
+	pinctrl-names = "default";
+};
+
+&twai {
+	pinctrl-0 = <&twai_default>;
+	pinctrl-names = "default";
+	bus-speed = <125000>;
+};
```
Comment (sylvioalves) [inline boards/xtensa/heltec_wireless_stick_lite_v3/heltec_wireless_stick_lite_v3.dts:217]: Please, add the missing line at the end.

File: boards/xtensa/heltec_wireless_stick_lite_v3/heltec_wireless_stick_lite_v3.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  - watchdog
+  - entropy
+  - pwm
+  - dma
+  - lora
+testing:
+  ignore_tags:
+    - net
+    - bluetooth
+    - lora
```
Comment (sylvioalves) [inline boards/xtensa/heltec_wireless_stick_lite_v3/heltec_wireless_stick_lite_v3.yaml:23]: Any reason to remove `lora` from the supported features?

**Comment by sylvioalves** [comment]
> @sylvioalves I left the TWAI and I2C-0 nodes in the DTS, but disabled them, and added a comment indicating that they can be enabled in a board extension. I checked the DTSI for the ESP32-S3 SoC and there's a driver for TWAI. However, some comments in the driver's source code suggest that some stuff may be pending to implement(?). Is that what you meant w.r.t. the TWAI driver?

I meant that TWAI "should" be enabled by default in case the board has a proper CAN interface/transceiver. Yes, ESP32S3 has TWAI. Well, this is just my opinion. You can leave as is right now.

**Comment by sylvioalves** [comment]
@adolfogc, there are some compliance issues, PTAL.

**Comment by adolfogc** [comment]
@sylvioalves Thanks for the review! I will work on this as soon as possible.

File: boards/xtensa/heltec_wireless_stick_lite_v3/heltec_wireless_stick_lite_v3.yaml (hunk 8 above)
Comment (adolfogc) [reply boards/xtensa/heltec_wireless_stick_lite_v3/heltec_wireless_stick_lite_v3.yaml:23]: I was able to test the LoRa samples using two boards, one with the receive firmware and another with the send firmware. The latter works but the former blocks and doesn't report receiving any message.

**Comment by sylvioalves** [comment]
@adolfogc, do you plan to update this PR soon? FYI, this PR can only be merged till tomorrow (and this board will be in release v3.5.0). Otherwise just in 2~3 weeks.

File: boards/xtensa/heltec_wireless_stick_lite_v3/heltec_wireless_stick_lite_v3.yaml (hunk 8 above)
Comment (adolfogc) [reply boards/xtensa/heltec_wireless_stick_lite_v3/heltec_wireless_stick_lite_v3.yaml:23]: Hi @sylvioalves, I have repeated the test using the latest SDK and the master branch as of recently. It appears to work now --the receive firmware--, but the output is garbled, so maybe I need to tweak the config parameters for the radio (US902-928 frequency)?

```bash
*** Booting Zephyr OS build v3.5.0-rc1-170-g7fb2929e5ba0 ***
[00:00:00.297,000] <inf> lora_receive: Synchronous reception
?_?H??&??9Zi? (RSSI:0dBm, SNR:-19dBm): Received data: ??p?t?T???????H'?S?
```

I think the issue was fixed by the PR by @fabiobaltieri referenced in https://github.com/zephyrproject-rtos/zephyr/issues/62427

**Comment by adolfogc** [comment]
> @adolfogc, do you plan to update this PR soon? FYI, this PR can only be merged till tomorrow (and this board will be in release v3.5.0). Otherwise just in 2~3 weeks.

@sylvioalves, sorry I wasn't able to look into this until today, I've made various changes to address the points raised in your latest review

**Comment by MaureenHelm** [comment]
@adolfogc please fix the failed CI checks

**Comment by adolfogc** [comment]
hi @MaureenHelm, thank you, I have updated the PR with the required changes

**Comment by MaureenHelm** [comment]
@sylvioalves please revisit

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+References
+**********
+
+- `Heltec Wireless Stick Lite (v3) Pinout Diagram <https://resource.heltec.cn/download/Wireless_Stick_Lite_V3/HTIT-WSL_V3.png>`_
+- `Heltec Wireless Stick Lite (v3) Schematic Diagrams <https://resource.heltec.cn/download/Wireless_Stick_Lite_V3/HTIT-WSL_V3_Schematic_Diagram.pdf>`_
+- `ESP-IDF Programming Guide <https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-reference/index.html>`_
+- `esptool documentation <https://github.com/espressif/esptool/blob/master/README.md>`_
+- `OpenOCD ESP32 <https://github.com/espressif/openocd-esp32/releases>`_
+
+.. [1] https://heltec.org/project/wireless-stick-lite-v2/
```
Comment (kartben) [inline boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:306]: FWIW this URL is very confusing :)

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+   HelTec Wireless Stick Lite (V3) (Credit: Chengdu HelTec Automation Technology Co., Ltd.)
+
+Hardware
+********
+
+The main hardware features are:
+
+- ESP32-S3FN8 low-power MCU-based SoC (dual-core Xtensa® 32-bit LX7 microprocessor, five stage pipeline rack Structure, main frequency up to 240 MHz).
+- Semtech SX1262 LoRa node chip
+- Type-C USB interface with a complete voltage regulator, ESD protection, short circuit protection, RF shielding, and other protection measures (note: you need an USB-A to USB-C cable if you want power-up the board from USB).
```
Comment (kartben) [inline boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:25]:
```suggestion
- Type-C USB interface with a complete voltage regulator, ESD protection, short circuit protection, RF shielding, and other protection measures (note: you need an USB-A to USB-C cable if you want to power-up the board from USB).
```

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst [hunk 11]
```diff
@@ -0,0 +1,306 @@
...
+   :widths: auto
+
+   +--------+---------+-----------------------------+
+   | Header | Function| Description                 |
+   +========+=========+=============================+
+   | J2.1   | Ve      |                             |
+   +--------+---------+-----------------------------+
+   | J2.2   | GND     |                             |
+   +--------+---------+-----------------------------+
+   | J2.3   | TBD     |                             |
```
Comment (kartben) [inline boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:61]: Not sure "TBD" is the right term for these?

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,306 @@
...
+   to the board default configuration file.
+   ```
+   CONFIG_BOOTLOADER_MCUBOOT=y
+   ```
+
+Sysbuild
+========
+
+The sysbuild makes possible to build and flash all necessary images needed to
+bootstrap the board with the EPS32S3 SoC.
```
Comment (kartben) [inline boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:189]: Correct spelling would be "ESP32-S3" but it looks like this generic section of the documentation is typically just using the term "ESP32" for all the other ESP32 boards.

```suggestion
bootstrap the board with the EPS32-S3 SoC.
```

**Review by kartben — CHANGES_REQUESTED** [review]
From a documentation perspective, see a few comments from my side but, more importantly, please replace the JPG images with a black background with ones with either white or transparent (preferred) background. I would recommend you use .webp format as this will help quite a bit with reducing image size. You may also crop the image to get rid of the empty background. Thanks!

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst (hunk 9 above)
Comment (adolfogc) [reply boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:306]: I agree it is very confusing.Unfortunately, the url has the "v2" in the slug but the product page it links to is actually "v3".

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst (hunk 11 above)
Comment (adolfogc) [reply boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:61]: Ok, I think I'll leave it blank instead.

**Comment by adolfogc** [comment]
> From a documentation perspective, see a few comments from my side but, more importantly, please replace the JPG images with a black background with ones with either white or transparent (preferred) background. I would recommend you use .webp format as this will help quite a bit with reducing image size. You may also crop the image to get rid of the empty background. Thanks!

Hi @kartben, thank you for your feedback! I have applied your suggestions and made some other changes as per your review.

File: boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst (hunk 9 above)
Comment (kartben) [reply boards/xtensa/heltec_wireless_stick_lite_v3/doc/index.rst:306]: Ya I figure there was probably little you would be able to do about it :)

**Comment by adolfogc** [comment]
Rebased and squashed the latest commit to pass the CI checks.

**Review by dcpleung — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**